src/dmd/backend/cgelem.d

```diff
@@ -5746,4 +5784,22 @@ private elem *elToPair(elem *e)
     return e;
 }
 
+/******************************************
+ * Determine if `b` can be moved before `a` without disturbing
+ * order-of-evaluation semantics.
+ */
+
+private bool canHappenAfter(elem* a, elem* b)
```

Comment:
Would `canMoveAfter` be a better name?

Reply:
Perhaps, but the uses of it actually move `b` before `a`.

Reply:
So, `canMoveBefore` or `canPrepend`? `canHappenAfter` seems like a very arbitrary name.